### PR TITLE
Feature/org admin groups

### PR DESCRIPTION
**Pull request by giancorzo**
https://developertown.visualstudio.com/SIL%20International%20Scriptoria/_workitems/edit/29573

- [x] Add group
- [x] List group
- [x] Remove group
- [x] Form validations
- [x] Edit group
- [x] Add tests

groups

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+export function withDataActions(InnerComponent) {
+
+  class DataActionWrapper extends React.Component<IProps> {
+
+    createRecord = async (payload: GroupAttributes) => {
+
+      const { dataStore, organization } = this.props;
+
+      try {
+        await dataStore.update(
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx]: any reason not to use `update` or `create` from `@data`?

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      } catch (e) {
+        toast.error(e);
+      }
+    }
+
+    removeRecord = async (group: GroupResource) => {
+      const { dataStore } = this.props;
+
+      try {
+        await dataStore.update(
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx]: I _think_ `@data` exports a destroy method? not sure. it'd be re-exported from store-helpers

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/groups/list.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    return (
+      <div className='m-t-lg'>
+        {
+          groups.map((group,index) =>
+            <div
+              key={index}
+              className='
+                col flex align-items-center
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/groups/list.tsx]: is `col` doing anything?

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/groups/list.tsx]: (not familair)

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx (hunk 1 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx]: Yeah I had issues trying to reRender the group list. The list reRender didn't wait for create or update method to finish even though I put async everywhere.

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/groups/list.tsx (hunk 3 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/groups/list.tsx]: Removing that class

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx (hunk 1 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx]: but, update / create doesn't have anything to do with async. :-\
async should be used regardless :_\

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx (hunk 2 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/group/with-data-actions.tsx]: why's that?

**Pull request merged**